src/components/editor/Bindings/Store/create.ts

```diff
@@ -0,0 +1,136 @@
+import { getDraftSpecsByCatalogName } from 'api/draftSpecs';
+import { getLiveSpecsByCatalogName } from 'api/liveSpecsExt';
+import { BindingsEditorState } from 'components/editor/Bindings/Store/types';
+import { CollectionData } from 'components/editor/Bindings/types';
+import produce from 'immer';
+import { isEmpty } from 'lodash';
+import { BindingsEditorStoreNames } from 'stores/names';
+import { devtoolsOptions } from 'utils/store-utils';
+import create, { StoreApi } from 'zustand';
+import { devtools, NamedSet } from 'zustand/middleware';
+
+const evaluateCollectionData = async (
+    draftId: string | null,
+    catalogName: string
+): Promise<CollectionData | null> => {
+    let draftSpecResponse = null;
+
+    if (draftId) {
+        draftSpecResponse = await getDraftSpecsByCatalogName(
+            draftId,
+            catalogName,
+            'collection'
+        );
+    }
+
+    if (draftSpecResponse && !isEmpty(draftSpecResponse.data)) {
+        return { spec: draftSpecResponse.data[0].spec, belongsToDraft: true };
+    } else {
+        const liveSpecResponse = await getLiveSpecsByCatalogName(
+            catalogName,
+            'collection'
+        );
+
+        return isEmpty(liveSpecResponse.data)
+            ? null
+            : { spec: liveSpecResponse.data[0].spec, belongsToDraft: false };
+    }
+};
+
+const getInitialStateData = (): Pick<
+    BindingsEditorState,
+    'collectionData' | 'schemaUpdateErrored' | 'schemaUpdated'
+> => ({
+    collectionData: null,
+    schemaUpdateErrored: false,
+    schemaUpdated: true,
+});
+
+const getInitialState = (
+    set: NamedSet<BindingsEditorState>,
+    get: StoreApi<BindingsEditorState>['getState']
+): BindingsEditorState => ({
+    ...getInitialStateData(),
+
+    initializeCollectionData: (currentCollection, persistedDraftId) => {
+        const { setCollectionData } = get();
+
+        if (currentCollection) {
+            evaluateCollectionData(persistedDraftId, currentCollection).then(
```

Comment:
There is a lot of async code in this PR and got me thinking.... we need to make a call on how we handle async code and returns. I kinda feel like we might want to always return promises/async stuff from any function call so that response is there if needed.

I think we'll want to make a call on this fairly soon.